designs/images/snowmachineconfig-controller-container-components.puml

```diff
@@ -0,0 +1,24 @@
+@startuml snowmachineconfig-controller-container-components
+!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Component.puml
+
+title Component diagram for SnowMachineConfig controller
+
+Person(user, "EKS-A user", "Manages EKS-A clusters")
+Container(kubeApiServer, "Kubernetes API server", "Go", "Provides read/write access to the kubernetes API objects stored in the etcd server and allows to subscribe to events when they change")
+
+Container_Boundary(snowMachineConfigController, "Snow MachineConfig Controller") {
+    Component(snowMachineConfigReconciler, "SnowMachineConfig Reconciler", "K8s controller-runtime Reconciler", "Runs validations for SnowMAchineConfig that require external API calls (eg. Snow endpoint)")
+    Component(awsCli, "AWS API client", "awscli - Python")
+
+    Rel(snowMachineConfigReconciler, awsCli, "Uses")
+}
+
+Component_Ext(snowAPI, "Snow API endpoint", "AWS?")
+Rel(awsCli, snowAPI, "Makes API calls", "JSON/HTTPS ??")
+
+Rel(user, kubeApiServer, "Writes EKS-A API objects", "JSON/HTTPS")
+
+Rel(snowMachineConfigController, kubeApiServer, "Subscribes to API events", "JSON/HTTPS")
```

Comment:
Similar to a previous comment, arrows from the subject boundary should really be coming from a component within the boundary so we understand _what performs the communication_.

Reply:
Same answer. I will address this with the same solution we agreed for the other controller